jax/_src/test_multiprocess.py

```diff
@@ -99,6 +100,18 @@ def _main(argv):
   if num_processes is None:
     raise ValueError("num_processes must be set")
   gpus_per_process = _GPUS_PER_PROCESS.value
+  # Get the number of GPUs visible to this process without initializing the runtime
+  if cuda_versions is not None:
+    local_device_count = cuda_versions.cuda_device_count()
+    if num_processes * gpus_per_process > local_device_count:
+      print(
```

Comment:
Should this be a `warnings.warn`?

Reply:
I think that might be too much? It might then need to be suppressed to avoid it being a hard error in `pytest` ("error" in https://github.com/jax-ml/jax/blob/7c394f41574b1f669cbb6a47fd03bf1925a19444/pyproject.toml#L67-L87).

If I understand correctly, the `multiaccelerator` tag is supposed to mean "it is worth exposing multiple GPUs to this test if you have them", so it should not be an error to run these tests with only a single GPU. There are many other examples of test cases being quietly (no warning) skipped if there are not enough visible devices - I was aiming for consistency.